File: trunk/EngineUAB/Code/Videogame/Characters/Player/AnimationStates/PlayerAnimationAttack3State.cpp
#include "PlayerAnimationAttack3State.h"
#include "StatesMachine\Telegram.h"
#include "RenderableObjects\AnimatedModel\AnimatedInstanceModel.h"

#if defined _DEBUG
#include "Memory\MemLeaks.h"
#endif

CPlayerAnimationAttack3State::CPlayerAnimationAttack3State( CCharacter* _pCharacter,const std::string &_Name )
	: CState(_pCharacter, _Name)
{

}

CPlayerAnimationAttack3State::~CPlayerAnimationAttack3State()
{

}

void CPlayerAnimationAttack3State::OnEnter( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("idle");
	_pCharacter->GetAnimatedModel()->BlendCycle(l_iAnimID, 0.1f);

	l_iAnimID = _pCharacter->GetAnimationID("attack3");
	_pCharacter->GetAnimatedModel()->ExecuteAction(l_iAnimID, 0.1f);
}

void CPlayerAnimationAttack3State::Execute( CCharacter* _pCharacter, float _fElapsedTime )
{

}

void CPlayerAnimationAttack3State::OnExit( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("idle");
	_pCharacter->GetAnimatedModel()->ClearCycle(l_iAnimID, 0.3f);
}

bool CPlayerAnimationAttack3State::OnMessage( CCharacter* _pCharacter, const STelegram& _Message )
{
	return false;
}

File: EngineUAB/Run/Videogame/Data/Scripts/Game/Player/player_animation_attack3_state.lua
class 'CPlayerAnimationAttack3State' (CState)
	function CPlayerAnimationAttack3State:__init(name) 
		CState.__init(self, name)
	end

	function CPlayerAnimationAttack3State:OnEnter(_CCharacter)
		if not ( _CCharacter == nil ) then
			num = _CCharacter:get_animation_id("attackfinal3")
			_CCharacter:get_animation_model():blend_cycle( num, 0.1 )
		
			--num = _CCharacter:get_animation_id("attack3")
			--_CCharacter:get_animation_model():execute_action( num, 0.1 )
		end
	end
	
	function CPlayerAnimationAttack3State:Execute(_CCharacter, _elapsed_time)
	end
	
	function CPlayerAnimationAttack3State:OnExit(_CCharacter)
		num = _CCharacter:get_animation_id("attackfinal3")
		_CCharacter:get_animation_model():clear_cycle( num, 0.3 )
	end
	
	function CPlayerAnimationAttack3State:OnMessage(_CCharacter, _Msg)
		return false
	end
	
	function CPlayerAnimationAttack3State:__Finalize()
	end